src/Microsoft.Identity.Client/Internal/Requests/SilentRequest.cs

```diff
@@ -58,26 +58,21 @@ internal override async Task<AuthenticationResult> ExecuteAsync(CancellationToke
                     "Token cache is set to null. Silent requests cannot be executed.");
             }
 
-            // Look for access token.
-            var msalAccessTokenItem =
-                await TokenCache.FindAccessTokenAsync(AuthenticationRequestParameters).ConfigureAwait(false);
+            MsalAccessTokenCacheItem msalAccessTokenItem = null;
 
-            MsalIdTokenCacheItem msalIdTokenItem = null;
+            // Look for access token
+            if (!ForceRefresh)
+            {
+                msalAccessTokenItem =
+                    await TokenCache.FindAccessTokenAsync(AuthenticationRequestParameters).ConfigureAwait(false);
+            }
```

Comment:
I moved the logic to check for ForceRefresh = true so that we circumvent the cache lookup. Previously, even if the dev passes in true, we still
look up the AT in the cache. Find AT lookup happening on [line 63][https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/dev/src/Microsoft.Identity.Client/Internal/Requests/SilentRequest.cs#L63](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/dev/src/Microsoft.Identity.Client/Internal/Requests/SilentRequest.cs#L63)) and check for ForceRefresh on [line 74](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/dev/src/Microsoft.Identity.Client/Internal/Requests/SilentRequest.cs#L74)

This later causes a [MultipleTokensMatchedError](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/dev/src/Microsoft.Identity.Client/TokenCache.cs#L426) because the look-up is happening when it should not.